config/routes.rb

```diff
@@ -177,6 +177,11 @@
   resources :partnerships, only: %i[index create show], param: :option
   resources :display_ad_events, only: [:create]
   resource :pro_membership, path: :pro, only: %i[show create update]
+  resources :user_blocks do
+    get ":blocked_id", on: :collection, to: "user_blocks#show"
+    post ":blocked_id", on: :collection, to: "user_blocks#create"
+    delete ":blocked_id", on: :collection, to: "user_blocks#destroy"
```

Comment:
I think these are going to conflict with the default routes, why did you add them instead of using the default ones?

This is how `rails routes` looks now:

```text
                                                                       GET      /user_blocks/:blocked_id(.:format)                                                user_blocks#show
                                                                       POST     /user_blocks/:blocked_id(.:format)                                                user_blocks#create
                                                                       DELETE   /user_blocks/:blocked_id(.:format)                                                user_blocks#destroy
                                                           user_blocks GET      /user_blocks(.:format)                                                            user_blocks#index
                                                                       POST     /user_blocks(.:format)                                                            user_blocks#create
                                                        new_user_block GET      /user_blocks/new(.:format)                                                        user_blocks#new
                                                       edit_user_block GET      /user_blocks/:id/edit(.:format)                                                   user_blocks#edit
                                                            user_block GET      /user_blocks/:id(.:format)                                                        user_blocks#show
                                                                       PATCH    /user_blocks/:id(.:format)                                                        user_blocks#update
                                                                       PUT      /user_blocks/:id(.:format)                                                        user_blocks#update
                                                                       DELETE   /user_blocks/:id(.:format)                                                        user_blocks#destroy
```

as you can see the three new ones, with the parameter called `blocked_id` point to the same actions.

If your goal is to rename the default parameter from `id`, to `blocked_id` you can do this:

```ruby
resources :user_blocks, param: :blocked_id, only: $%[show create destroy]
```

which generates:

```text
                                                           user_blocks POST     /user_blocks(.:format)                                                            user_blocks#create
                                                            user_block GET      /user_blocks/:blocked_id(.:format)                                                user_blocks#show
                                                                       DELETE   /user_blocks/:blocked_id(.:format)                                                user_blocks#destroy
```

you don't need to declare the parameters for the create action in the routes

Reply:
Exactly what I was trying to do! TIL; thanks!